Description: Fix spelling issues found by Lintian
Author: Axel Beckert <abe@debian.org>
Forwarded: no

--- a/client/xymonclient-sunos.sh
+++ b/client/xymonclient-sunos.sh
@@ -109,7 +109,7 @@ global|Global)
 esac
 echo "[swap]"
 /usr/sbin/swap -s
-# dont report the swaplist in a non-global zone because it will cause the
+# don't report the swaplist in a non-global zone because it will cause the
 # server client module to miscalculate memory
 case $ZTYPE in
 global|Global)
--- a/xymond/do_alert.c
+++ b/xymond/do_alert.c
@@ -413,7 +413,7 @@ void send_alert(activealerts_t *alert, F
 			if (!rpt) continue;	/* Happens for e.g. M_IGNORE recipients */
 
 			/* 
-			 * Update alertcount here, because we dont want to hit an UNMATCHED
+			 * Update alertcount here, because we don't want to hit an UNMATCHED
 			 * rule when there is actually an alert active - it is just suppressed
 			 * for this run due to the REPEAT setting.
 			 */
--- a/xymond/xymond.c
+++ b/xymond/xymond.c
@@ -3175,7 +3175,7 @@ int match_host_filter(void *hinfo, hostf
 			break;
 
 		  case FILTER_TAG:
-			if (!matchontests) break;	/* When processing a xymondboard, we dont want to return entries that have a test in hosts.cfg but no test result */
+			if (!matchontests) break;	/* When processing a xymondboard, we don't want to return entries that have a test in hosts.cfg but no test result */
 			matched = 0;
 			val = xmh_item_walk(hinfo);
 			while (val) {
