Description: Fix spelling issues found by Lintian
Author: Axel Beckert <abe@debian.org>
Forwarded: no

Index: xymon/client/xymonclient-sunos.sh
===================================================================
--- xymon.orig/client/xymonclient-sunos.sh	2017-01-04 18:37:47.147672839 +0100
+++ xymon/client/xymonclient-sunos.sh	2017-01-04 18:37:47.139672853 +0100
@@ -109,7 +109,7 @@
 esac
 echo "[swap]"
 /usr/sbin/swap -s
-# dont report the swaplist in a non-global zone because it will cause the
+# don't report the swaplist in a non-global zone because it will cause the
 # server client module to miscalculate memory
 case $ZTYPE in
 global|Global)
Index: xymon/xymond/do_alert.c
===================================================================
--- xymon.orig/xymond/do_alert.c	2017-01-04 18:37:47.147672839 +0100
+++ xymon/xymond/do_alert.c	2017-01-04 18:37:47.143672845 +0100
@@ -420,7 +420,7 @@
 			if (!rpt) continue;	/* Happens for e.g. M_IGNORE recipients */
 
 			/* 
-			 * Update alertcount here, because we dont want to hit an UNMATCHED
+			 * Update alertcount here, because we don't want to hit an UNMATCHED
 			 * rule when there is actually an alert active - it is just suppressed
 			 * for this run due to the REPEAT setting.
 			 */
Index: xymon/xymond/xymond.c
===================================================================
--- xymon.orig/xymond/xymond.c	2017-01-04 18:37:47.147672839 +0100
+++ xymon/xymond/xymond.c	2017-01-04 18:37:47.143672845 +0100
@@ -3190,7 +3190,7 @@
 			break;
 
 		  case FILTER_TAG:
-			if (!matchontests) break;	/* When processing a xymondboard, we dont want to return entries that have a test in hosts.cfg but no test result */
+			if (!matchontests) break;	/* When processing a xymondboard, we don't want to return entries that have a test in hosts.cfg but no test result */
 			matched = 0;
 			val = xmh_item_walk(hinfo);
 			while (val) {
