Description: Fix multiple CGI-related regressions in 4.3.26
Origin: http://lists.xymon.com/pipermail/xymon/2016-March/043132.html
Author: J.C. Cleaver <cleaver@terabithia.org>

Index: xymon/lib/cgi.c
===================================================================
--- xymon.orig/lib/cgi.c	2016-02-26 18:19:04.978134697 +0100
+++ xymon/lib/cgi.c	2016-03-04 14:59:03.976168123 +0100
@@ -41,6 +41,16 @@
 	return result;
 }
 
+int cgi_ispost()
+{
+	char *method = getenv("REQUEST_METHOD");
+	if (method && (strcasecmp(method, "POST") == 0)) {
+		cgi_method = CGI_POST; /* might as well set here */
+		return 1;
+	}
+	return 0;
+}
+
 cgidata_t *cgi_request(void)
 {
 	char *method = NULL;
@@ -267,7 +277,7 @@
 	else if (strncmp(str, "criticaleditor", 14) == 0) csppol = strdup("script-src 'self'; connect-src 'self'; form-action 'self';");
 	else if (strncmp(str, "svcstatus-trends", 16) == 0) csppol = strdup("script-src 'self' 'unsafe-inline'; connect-src 'self'; form-action 'self'; sandbox allow-forms allow-scripts;");
 	else if (strncmp(str, "svcstatus-info", 14) == 0) csppol = strdup("script-src 'self' 'unsafe-inline'; connect-src 'self'; form-action 'self'; sandbox allow-forms allow-same-origin allow-scripts allow-modals allow-popups;");
-	else if (strncmp(str, "svcstatus", 9) == 0) csppol = strdup("script-src 'self'; connect-src 'self'; form-action 'self'; sandbox allow-forms;");
+	else if (strncmp(str, "svcstatus", 9) == 0) csppol = strdup("script-src 'self'; connect-src 'self'; form-action 'self'; sandbox allow-forms allow-same-origin;");
 	else if (strncmp(str, "historylog", 10) == 0) csppol = strdup("script-src 'self'; connect-src 'self'; form-action 'self'; sandbox allow-forms;");
 	else {
 		errprintf(" csp_header: page %s not listed, no CSP returned\n", str);
Index: xymon/lib/cgi.h
===================================================================
--- xymon.orig/lib/cgi.h	2016-02-26 18:19:04.978134697 +0100
+++ xymon/lib/cgi.h	2016-03-04 14:59:03.976168123 +0100
@@ -22,6 +22,7 @@
 extern enum cgi_method_t cgi_method;
 
 extern char *cgi_error(void);
+extern int cgi_ispost(void);
 extern cgidata_t *cgi_request(void);
 extern char *csp_header(const char *pagename); 
 extern int cgi_refererok(char *expected); 
Index: xymon/web/ackinfo.c
===================================================================
--- xymon.orig/web/ackinfo.c	2016-02-26 18:19:04.982134677 +0100
+++ xymon/web/ackinfo.c	2016-03-04 14:59:03.976168123 +0100
@@ -97,10 +97,14 @@
 	{
 	    char cgisource[1024]; char *p;
 	    p = csp_header("ackinfo"); if (p) fprintf(stdout, "%s", p);
-	    snprintf(cgisource, sizeof(cgisource), "%s/%s", xgetenv("SECURECGIBINURL"), "criticalview");
+	    snprintf(cgisource, sizeof(cgisource), "%s/%s", xgetenv("CGIBINURL"), "svcstatus");
 	    if (!cgi_refererok(cgisource)) {
-		fprintf(stdout, "Location: %s.sh?\n\n", cgisource);
-		return 0;
+		snprintf(cgisource, sizeof(cgisource), "%s/%s", xgetenv("CGIBINURL"), "criticalview");
+		if (!cgi_refererok(cgisource)) {
+			errprintf("ackinfo POST that is not coming from criticalview or svcstatus (referer=%s). Ignoring.\n", textornull(getenv("HTTP_REFERER")) );
+			fprintf(stdout, "Location: %s.sh?\n\n", cgisource);
+			return 0;
+		}
 	    }
 	}
 
Index: xymon/web/criticaleditor.c
===================================================================
--- xymon.orig/web/criticaleditor.c	2016-02-26 18:19:04.982134677 +0100
+++ xymon/web/criticaleditor.c	2016-03-04 14:59:03.976168123 +0100
@@ -391,7 +391,7 @@
 	redirect_cgilog("criticaleditor");
 
         /* We only want to accept posts from certain pages */
-	{
+	if (cgi_ispost()) {
 		char cgisource[1024]; char *p;
 		p = csp_header("criticaleditor"); if (p) fprintf(stdout, "%s", p);
 		snprintf(cgisource, sizeof(cgisource), "%s/%s", xgetenv("SECURECGIBINURL"), "criticaleditor");
