--- a/build/test-pathmax.c
+++ b/build/test-pathmax.c
@@ -8,6 +8,9 @@
 
 #ifndef PATH_MAX
 	res = pathconf("/", _PC_PATH_MAX);
+	if (res == -1) {
+		res = 4096;
+	}
 	printf("#define PATH_MAX %ld\n", res);
 #endif
 
--- a/lib/url.h
+++ b/lib/url.h
@@ -11,6 +11,20 @@
 #ifndef __URL_H__
 #define __URL_H__
 
+#ifndef MAXPATHLEN
+# ifdef PATH_MAX
+#  define MAXPATHLEN PATH_MAX
+# elif defined(MAX_PATH)
+#  define MAXPATHLEN MAX_PATH
+# elif defined(_MAX_PATH)
+#  define MAXPATHLEN _MAX_PATH
+# elif defined(CCHMAXPATH)
+#  define MAXPATHLEN CCHMAXPATH
+# else
+#  define MAXPATHLEN 1024
+# endif
+#endif
+
 extern int obeybbproxysyntax;
 
 typedef struct urlelem_t {
--- /dev/null
+++ b/build/Makefile.GNU
@@ -0,0 +1,32 @@
+# Xymon compile-time settings for LINUX systems
+
+OSDEF = -DLINUX
+
+# NETLIBS: None needed on Linux
+NETLIBS =
+
+# Compile flags for normal build
+CC = gcc
+CFLAGS += -Wall -Wno-unused -Wno-pointer-sign -D_REENTRANT $(LFSDEF) $(OSDEF)
+
+ifndef PKGBUILD
+RPATH = -Wl,--rpath,
+endif
+
+# Compile flags for debugging
+# CFLAGS = -g -DDEBUG -Wall -D_REENTRANT $(LFSDEF) $(OSDEF)
+
+# For profiling
+# CFLAGS = -g -pg -Wall -D_REENTRANT $(LFSDEF) $(OSDEF)
+# Note: Do 'export GMON_OUT_PREFIX=/var/tmp/FILENAME' to save profiling output in /var/tmp
+# Use gprof to analyse
+
+# By default, Xymon uses a static library for common code.
+# To save some diskspace and run-time memory, you can use a
+# shared library by un-commenting this.
+# XYMONLIBRARY=libxymon.so
+
+# Mail program: This must support "CMD -s SUBJECT ADDRESS" to send out a mail with a subject
+# Typically, this will be "mail" or "mailx"
+MAILPROGRAM="mail"
+
